web/models/campaign.ex

```diff
@@ -15,17 +15,36 @@ defmodule Jingle.Campaign do
     belongs_to :sponsor, Jingle.Sponsor
     belongs_to :podcast, Jingle.Podcast
     has_many :creatives, Jingle.Creative
-    timestamps()
+    timestamps(type: :utc_datetime)
   end
 
   @doc """
   Builds a changeset based on the `struct` and `params`.
   """
   def changeset(struct, params \\ %{}) do
     struct
-    |> cast(params, [:sponsor_id, :podcast_id, :start_date, :end_date, :due_date, :original_copy, :edited_copy, :must_say, :notes, :approved, :zone])
-    |> validate_required([:start_date, :end_date, :original_copy, :zone])
+    |> cast(utc_compatible_dates(params), [:sponsor_id, :podcast_id, :start_date, :end_date, :due_date, :original_copy, :edited_copy, :must_say, :notes, :approved, :zone])
+    |> validate_required([:start_date, :end_date, :due_date, :original_copy, :zone])
     |> assoc_constraint(:sponsor)
     |> assoc_constraint(:podcast)
   end
+
+  defp utc_compatible_dates(params) do
+    date_format = "{YYYY}-{0M}-{0D}"
```

Comment:
So it seems like this is either going to (1) parse a plain date-string to a `%NaiveDateTime`, or (2) rescue and pass the string through to Ecto's built in parser.

Is the advantage to (2) just to let the Ecto changeset handle passing any "invalid date" errors on through the controller?

Reply:
made some changes, but re: 2) -- yes